### DISCUSSION TITLE
Interface setting on CUBIT mesh

### DISCUSSION THREAD

**Discussion by styyokuda** [idaholab/moose#22757] [Q&A General] [answered]
All,
I have a mesh generated by CUBIT (please see the attached sketch).
I would like to add interfaces on the left, right, top, and bottom sides of Block 2.
All I know is to use "SideSetsBetweenSubdomainsGenerator" which generates one interface around Block 2.

CUBIT generated SideSets for these left, right, top and bottom boundaries.
Please help me to generate separately the interfaces on the left, right, top, and bottom sides of Block 2.
 
Sincerely,
S. Thomas
[Interfaces_two_variables_two_blocks.pdf](https://github.com/idaholab/moose/files/10045742/Interfaces_two_variables_two_blocks.pdf)

**Comment by GiudGiud** [comment, marked as answer]
Hello

You may use the SidesetsAroundSubdomainGenerator with a normal parameter since the sidesets are all with a different normal.
https://mooseframework.inl.gov/moose/source/meshgenerators/SideSetsAroundSubdomainGenerator.html

You will need to use as many of these objects, chained using the input parameter, as there are sidesets to add.

Guillaume

**Reply by styyokuda** [reply]
Hi Guillaume,
I am using "InterfaceDiffusion" of "InterfaceKernels" whose interface is generated by "SideSetsBetweenSubdomainsGenerator" (test/tests/interfacekernels/hybrid/interface.i).

"SidesetsAroundSubdomainGenerator" seems to generate the same interface as "SideSetsBetweenSubdomainsGenerator" does.

Please confirm that is true.

Sincerely,
S. Thomas

**Reply by GiudGiud** [reply]
you ll have to use the normal parameter to distinguish each surface

**Reply by styyokuda** [reply]
Thank you, Guillaume.
Both "SidesetsAroundSubdomainGenerator" and "SideSetsBetweenSubdomainsGenerator" produce the same results for the interface model.

S. Thomas
